src/sema/tests/mod.rs

```diff
@@ -128,3 +139,86 @@ fn test_statement_reachable() {
         assert_eq!(test_case.reachable(), expected);
     }
 }
+
+#[test]
+fn constant_overflow_checks() {
```

Comment:
All code paths should be tested, so you need tests for all bit widths and all arithmetic operations.

Reply:
I think testing for all bit width is not needed, bits of a type are retrieved through a BigInt function (bits()). see: https://github.com/salaheldinsoliman/solang/blob/2c9c38ead19ed472f3054dc7f2e2470f5c92d919/src/sema/eval.rs#L353
So I stuck for uint8 and int8 for the sake of readability.
But I agree this test needs more code path

Reply:
You still need tests for other bit widths. Had you tested for `u64` here, you would have noticed that using `to_u32` as I have pointed out would not work.

The purpose of writing tests is to ensure everything will work as expected and we do not face regressions. If you don't write tests for other bit widths, someone might change your code that makes overflow detection for, let's say, `int256` stop working. As we don't have tests for that, this regression will be unnoticed. Tests are a reference for ensuring that new changes do not break any functionality.

Reply:
Why has this been marked as resolved @salaheldinsoliman ?

Reply:
I added another function that tests for other types but with less test cases